Update and rename 2024-02-15-waku-network-tech-overview.mdx to 2024-0… (#27)
waku-org/blog.waku.org: posts/2024-02-19-waku-network-tech-overview.mdx

		
- DOS protection: ensuring that the network cannot be flooded with messages, leading users who have fewer resources to be booted off the network.
- Network sustainability and incentivisation: How do we ensure enough resources in the network to enable devices such as browsers and mobile devices to access the network?
- Decentralisation: Waku needs to be decentralised at several levels to enable those properties. How do we maintain a fair level of decentralisation to ensure those properties remain over time?
		
## The Waku Network
		
How does The Waku Network help deliver the properties above?
		
Let us review the different Waku protocols that are combined in the network and how they enable us to achieve this feat.
		
### Peer Discovery
		
For any peer-to-peer system to be reliable and decentralised, there must be a mechanism to find new peers or nodes in said system, usually called _peer discovery_.
		
Waku uses discv5, similar to Ethereum. Minor enhancements to ENR have been done to enable the Waku node to advertise:
		
- the shards they are operating in (see [message routing - sharding](#message-routing--sharding))

> 	- the protocol they have enabled
> 	- alternative multiaddr they may have, e.g., for the browser to connect to the said node via WebSocket.

Reviewer: ```suggestion
- the protocol they have enabled
- alternative multiaddr they may have, e.g., for the browser to connect to the said node via WebSocket.
```
extra tabs

		
This is done with RLN, or Rate Limiting Nullifier, which limits the rate of messages a given publisher sends.
It is currently set to 1 msg/s. Coupled with a maximum message size (150kB) and a maximum number of publishers (80k, TBD), we can assume a maximum bandwidth usage per shard (around 10Mbps).
		
Rate-limiting publishers in a censorship-resistant and private manner is difficult; this is why we are using zero-knowledge technology:
		
1. Users push their RLN credentials to a smart contract (currently on Ethereum Sepolia testnet).
2. Nodes keep track of all registered memberships on the contract.
3. When sending a message, the user attaches an RLN proof to the message with the current epoch (timestamp in seconds).
4. Nodes can verify the proof without being able to correlate the user's Ethereum address (used on the smart contract) with the message being sent, preserving anonymity.
5. If a user attempts to send more than 1 msg/s, nodes can detect this and drop the message surplus or spam, protecting the network.

> ### Servicing Mostly-offline and Resource-restricted Devices

Reviewer: I believe there was some acid info guideline that we should only capitalized first word in title?

Finally, how is Waku useful to resource-restricted devices, such as mobile phones and web browsers?

> Waku defines a number of [request-response protocols (https://rfc.vac.dev/spec/10/#requestreply-domain) to enable such devices to access the Waku network without having to be always online or consume extensive amounts of bandwidth, i.e., without participating in the Waku Relay network.

Reviewer: ```suggestion
Waku defines a number of [request-response protocols](https://rfc.vac.dev/spec/10/#requestreply-domain) to enable such devices to access the Waku network without having to be always online or consume extensive amounts of bandwidth, i.e., without participating in the Waku Relay network.
```

> The light push protocol (links to docs) enables a _light client_ to send a message to be forwarded to the Waku Relay network, with acknowledgement of reception from the remote node.

Reviewer: ```suggestion
The [light push protocol](https://docs.waku.org/learn/concepts/protocols#light-push) enables a _light client_ to send a message to be forwarded to the Waku Relay network, with acknowledgement of reception from the remote node.
```

> The filter protocol enables a _light client_ to subscribe to a remote peer and only request a subset of messages instead of all messages transmitted on a shard.

Reviewer: ```suggestion
The [filter protocol](https://docs.waku.org/learn/concepts/protocols/#filter) enables a _light client_ to subscribe to a remote peer and only request a subset of messages instead of all messages transmitted on a shard.
```

		
Finally, the store protocol enables _light clients_ and _relay nodes_ to retrieve historical messages that may have been missed.
		
## The Waku Network's Value Proposition
		
While we have defined the desired properties and the technology, it is important to understand the potential use cases of the Waku Network. This [issue](https://github.com/waku-org/docs.waku.org/issues/134) currently describes several USPs (Unique Value Propositions), and we will be documenting the subject further. Some noteworthy use cases are, beyond building any machine-to-machine or human-to-human communication app:
		
- No-Infra DApps: combining various decentralised technologies (Waku, Ethereum, IPFS) to deploy a DApp without paying a centralised hosting provider.
- Censorship-resistant access to a decentralised network for light clients: using the request-response protocols to enable light clients to access your peer-to-peer network without relying on a centralised, censorable web gateway.
- Signal network: Use The Waku Network to find other peers and negotiate specific parameters to form your own peer-to-peer, Waku or not, network with different rules (higher rate limit, etc.).

> ## Are we _blanlk_ yet?

Reviewer: ```suggestion
## Are we _blank_ yet?
```

		
We have described Waku's desired properties and how the Waku Network delivers on those. Does it mean Waku is censorship-resistant, private, sustainable, and scalable? *Not quite.*
		
Bootstrapping is a component of all peer-to-peer networks we did not address: how does a new node find other nodes in the network? We use Ethereum technology for bootstrapping (ENR + DNS Discovery). However, this technology could be more decentralised. We intend to improve this potential around the end of 2024 or 2025.
		

> In general, the network needs to be decentralised to achieve the desired properties. While some of the technology enables such decentralisation (and we are working on improving the part that does not), this problem has a social component. If the Waku team is the only one running nodes, then inherently, the network cannot be considered decentralised, and hence censorship-resistant, etc. To solve this, we need to push for the adoption of the Waku network so we can build a good base of node operators and developers that run nodes without depending on the Waku team to do so.

Reviewer: ```suggestion
In general, the network needs to be decentralised to achieve the desired properties. While some of the technology enables such decentralisation (and we are working on improving the parts that do not), this problem has a social component. If the Waku team is the only one running nodes, then inherently, the network cannot be considered decentralised, and hence censorship-resistant, etc. To solve this, we need to push for the adoption of the Waku network so we can build a good base of node operators and developers that run nodes without depending on the Waku team to do so.
```

		
We increased This effort last year and will continue this year.
		

> Another aspect is providing monetary incentivisation to node operators to run nodes so the network can become self-sustainable. We do not have such a protocol; we are building the first PoC.

Reviewer: ```suggestion
Another aspect is providing monetary incentivisation to node operators to run nodes so the network can become self-sustainable. We do not have such a protocol; we are [building the first PoC](https://github.com/vacp2p/rfc/tree/master/content/docs/rfcs/73).
```
